.github/workflows/container-build.yml

```diff
@@ -59,5 +59,21 @@ jobs:
       - name: creds
         run: gcloud auth configure-docker --quiet
 
+      - name: Formatted labels
+        id: labels
+        run: |
+          FORMATED_LABELS="--image-label commit-hash=$GITHUB_SHA"
+
+          BRANCH_NUMBER=$(gh pr list --state all --search "sha:$GITHUB_SHA" --label "breaking-change" | awk '{print $1}')
```

Comment:
Can we check if we need to create a PAT for this or if the default token is fine? We sometimes hit rate limits when querying GitHub. I’d rather not manage another token but just wanna check. 

Cc @cpanato

Reply:
One other idea is to change this to fire on PR closure, like what’s documented in  https://stackoverflow.com/questions/71523153/github-actions-on-pull-request-merged-with-specific-labels  

You could then checkout from HEAD based on the git sha.

Reply:
Eh, that might not work - commit shas will differ after being squashed to merge to HEAD.

Reply:
I think it should be fine as I tested this same action in my fork and it worked properly. https://github.com/javanlacerda/fulcio/actions/runs/10997017464/job/30531560844. Just hadn't the authorization to push the image, what is expected.

Reply:
i see this and will reply tomorrow (25/sept/2024) sorry !

Reply:
@cpanato could you take a look ?

Reply:
we need to add the permission  `pull-request: read`

Reply:
but this will also require that we add this label in the PRs, do we have a process documented for that?

Reply:
We don't have a document that cover all the process, but this label usage started by the necessity to detect builds that requires adicional configuration on the helm charts. Basically, this breaking change label says that this image should not be deployed automatically and requires manual review, checking the configurations. We decided to use the `breaking-change` name because we can generalize it for having blocking auto deployment and requiring adicional review for any kind of blocker.

Reply:
> We sometimes hit rate limits when querying GitHub

`gh pr list` does not use the "search" endpoint which I believe is the problematic one (since search rate limit is 30/minute instead of 5000/hour like others so temporary spikes can easily hit it) so I guess this is fine.

Reply:
if i run that locally it does not work, need to do a setup before

```
$ gh pr list --state all --search "sha:7920be28f7f58020c4da8ab9bb78554ad7cd67ea" --label "breaking-change"
X No default remote repository has been set for this directory.

please run `gh repo set-default` to select a default remote repository.
```

Reply:
ok, so maybe is a local thing, not related to the gh actions env. thanks for the clarification